### PR TITLE
Set up api prow jobs

### PR DESCRIPTION
**Pull request by chxchx**
Merge only after https://github.com/istio/api/pull/189

**Release note**:

```release-note
none
```

### CODE REVIEW

**Review by kyessenov — COMMENTED** [review]
/lgtm

File: prow/config.yaml [hunk 1]
```diff
@@ -26,6 +26,66 @@ presubmits:
...
+        securityContext:
+          privileged: true
+        env:
+        - name: GOOGLE_APPLICATION_CREDENTIALS
+          value: /etc/service-account/service-account.json
+        volumeMounts:
+        - name: service
+          mountPath: /etc/service-account
+          readOnly: true
+        - name: api-e2e-rbac-kubeconfig
```
Comment (yutongz) [inline prow/config.yaml]: Remove this part too if we don't need a cluster and code coverage for now

Comment (chxchx) [reply prow/config.yaml]: Done

**Review by sebastienvas — APPROVED** [review]

**Pull request merged**